src/main/java/org/cdc/gov/sdp/aphl/AphlS3Producer.java

```diff
@@ -0,0 +1,334 @@
+import org.apache.camel.Message;
+import org.apache.camel.WrappedFile;
+import org.apache.camel.component.aws.ec2.EC2Constants;
+import org.apache.camel.component.aws.s3.S3Configuration;
+import org.apache.camel.component.aws.s3.S3Constants;
+import org.apache.camel.component.aws.s3.S3Operations;
+import org.apache.camel.impl.DefaultProducer;
+import org.apache.camel.util.CastUtils;
+import org.apache.camel.util.FileUtil;
+import org.apache.camel.util.IOHelper;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.camel.util.URISupport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.services.cloudfront.model.InvalidArgumentException;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.AbortMultipartUploadRequest;
+import com.amazonaws.services.s3.model.AccessControlList;
+import com.amazonaws.services.s3.model.Bucket;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadRequest;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadResult;
+import com.amazonaws.services.s3.model.CopyObjectRequest;
+import com.amazonaws.services.s3.model.CopyObjectResult;
+import com.amazonaws.services.s3.model.DeleteBucketRequest;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadResult;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PartETag;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.StorageClass;
+import com.amazonaws.services.s3.model.UploadPartRequest;
+
+/**
+ * A Producer which sends messages to the Amazon Web Service Simple Storage
+ * Service <a href="http://aws.amazon.com/s3/">AWS S3</a>
+ */
+public class AphlS3Producer extends DefaultProducer {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AphlS3Producer.class);
+
+	private transient String s3ProducerToString;
+
+	public AphlS3Producer(final Endpoint endpoint) {
+		super(endpoint);
+	}
+
+	@Override
+	public void process(final Exchange exchange) throws Exception {
+
+		if (getConfiguration().isMultiPartUpload()) {
+			processMultiPart(exchange);
+		} else {
+			processSingleOp(exchange);
+		}
+	}
+
+	public void processMultiPart(final Exchange exchange) throws Exception {
+		File filePayload = null;
+		Object obj = exchange.getIn().getMandatoryBody();
+		// Need to check if the message body is WrappedFile
+		if (obj instanceof WrappedFile) {
+			obj = ((WrappedFile<?>) obj).getFile();
+		}
+		if (obj instanceof File) {
+			filePayload = (File) obj;
+		} else {
+			throw new InvalidArgumentException("aphl-s3: MultiPart upload requires a File input.");
+		}
+
+		ObjectMetadata objectMetadata = determineMetadata(exchange);
+		if (objectMetadata.getContentLength() == 0) {
```

Comment:
This class is again derived from the camel s3 components.  I think this is primarily a means to tell the s3 client how big the content is supposed to be to avoid buffering the entire thing in memory to figure it out.  That way the client can simply stream the data out to s3.